src/Game.cpp

```diff
@@ -39,17 +39,174 @@ static void sub_53A890(const char *args) {
+
+    char* skuVersion = NULL;
+
+    switch (global_SKUVersion) {
+    case 0:
+        skuVersion = "SLUS-20209";
+        break;
+    case 1:
+        skuVersion = "SLES-51054";
+        break;
+    case 2:
+        skuVersion = "SLPM-20209";
+        break;
+    default:
+        // SKU unrecognised
+        mc2_log_error("Unhandled sku");
+        return;
+    }
+
+    mc2_log_C("mcConfig:AutoDetectSku - product code '%s'", skuVersion);
+}
+
+// mc2: 0x0053B9B0
+// Set default resolution?
+void sub_53B9B0() {
+    glo_66315C = 0;
+    
+    if (sub_612E10("pc_480")) {
+        // 480 x 640
+        glo_66315C = 7;
+    } 
+    else if (sub_612E10("pc_720")) {
+        // 720 x 480
+        glo_66315C = 8;
+    }
+}
+
+int sub_5E1B50(const char* path) {
+    for (int i = 0; i < 10; ++i) {
+        char* language = global_LanguageList[i];
+
+        if (sub_627145(path, language)) {
+            return i;
+        }
+    }
+    return -1;
+}
+
+bool sub_539DE0(const char* path) {
+    for (int i = 0; i < 10; ++i) {
+        char* unk = glo_663160[i];
+
+        if (sub_627145(path, unk)) {
+            glo_6C2C60 = i;
+            return true;
+        }
+    }
+    mc2_log_fatal("mcConfig::SetMode() failed");
+    return false;
+}
+
+void sub_53A000() {
+    check_sku_version();
+    sub_53B9B0();
+
+    if (sub_612E10("unlockall")) {
+        glo_unlock_races = 1;
+        glo_unlock_cities = 1;
+        glo_unlock_vehicles = 1;
+        glo_unlock_customization = 1;
+        glo_unlock_abilities = 1;
+        glo_unlock_difficulty = 1;
+    }
+    else {
+        if (sub_612E10("unlockraces")) {
+            glo_unlock_races = 1;
+        }
+        if (sub_612E10("unlockcities")) {
+            glo_unlock_cities = 1;
+        }
+        if (sub_612E10("unlockvehicles")) {
+            glo_unlock_vehicles = 1;
+        }
+        if (sub_612E10("unlockcustomization")) {
+            glo_unlock_customization = 1;
+        }
+        if (sub_612E10("unlockabilities")) {
+            glo_unlock_abilities = 1;
+        }
+        if (sub_612E10("unlockinsanity")) {
+            glo_unlock_difficulty = 1;
+        }
+        if (sub_612E10("unlockdifficulty")) {
+            glo_unlock_difficulty = 1;
+        }
+    }
+    if (sub_612E10("lang")) {
+        const char* path;
+        sub_612EB0("lang", 0, &path);
+        int lang = sub_5E1B50(path);
+        if (lang == -1) {
+            mc2_log_error("Invalid language");
+            return;
+        }
+
+        mc2_log_C("Setting language to %s (%d)", global_LanguageList[lang], lang);
+        global_LanguageID = lang;
+    }
+
+    if (sub_612E10("fromconfigurator")) {
+        glo_6C2C64 = 1;
+    }
+
+    const char* unk;
+    if (sub_612EB0("mode", 0, &unk)) {
+        sub_539DE0(unk);
+    }
+}
+
+void sub_6134D0(const char* var8) {
+    char* destination = glo_860120;
+    memset(destination, 0, 256);
+    safe_strncpy(destination, var8, 255);
```

Comment:
But the 255 version is safer as the last character 256 will be definitely be a zero.